Explorer/Assets/DCL/SDKComponents/Tween/Systems/TweenUpdaterSystem.cs

```diff
@@ -0,0 +1,242 @@
+using CrdtEcsBridge.ECSToCRDTWriter;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Optimization.Pools;
+using DCL.SDKComponents.Tween.Components;
+using DCL.SDKComponents.Tween.Helpers;
+using DG.Tweening;
+using ECS.Abstract;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Unity.Groups;
+using ECS.Unity.Transforms.Components;
+using System.Collections.Generic;
+using UnityEngine;
+using static DCL.ECSComponents.EasingFunction;
+using static DG.Tweening.Ease;
+
+namespace DCL.SDKComponents.Tween.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [UpdateAfter(typeof(TweenLoaderSystem))]
+    [LogCategory(ReportCategory.TWEEN)]
+    [ThrottlingEnabled]
+    public partial class TweenUpdaterSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private const int MILLISECONDS_CONVERSION_INT = 1000;
+
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+        private List<DG.Tweening.Tween> transformTweens = new List<DG.Tweening.Tween>();
+        private Tweener tempTweener;
+        private readonly IComponentPool<SDKTweenComponent> tweenComponentPool;
+
+
+        private TweenUpdaterSystem(World world, IECSToCRDTWriter ecsToCRDTWriter, IComponentPool<SDKTweenComponent> tweenComponentPool) : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+            this.tweenComponentPool = tweenComponentPool;
+        }
+
+        protected override void Update(float t)
+        {
+            UpdateTweenSequenceQuery(World);
+            HandleEntityDestructionQuery(World);
+        }
+
+        public void FinalizeComponents(in Query query)
+        {
+            FinalizeComponentsQuery(World);
+        }
+
+        [Query]
+        [All(typeof(TweenComponent))]
+        private void FinalizeComponents(in Entity entity, ref CRDTEntity sdkEntity)
+        {
+            ecsToCRDTWriter.DeleteMessage<PBTweenState>(sdkEntity);
+            World.Remove<TweenComponent>(entity);
+        }
+
+        [Query]
+        [All(typeof(DeleteEntityIntention))]
+        private void HandleEntityDestruction(ref TweenComponent tweenComponent, ref CRDTEntity sdkEntity)
+        {
+            ecsToCRDTWriter.DeleteMessage<PBTweenState>(sdkEntity);
+            tweenComponentPool.Release(tweenComponent.SDKTweenComponent);
+        }
+
+        [Query]
+        [All(typeof(TweenComponent))]
+        private void UpdateTweenSequence(in Entity entity, ref TweenComponent tweenComponent, ref TransformComponent transformComponent, ref CRDTEntity sdkEntity)
+        {
+            var sdkTweenComponent = tweenComponent.SDKTweenComponent;
+
+            if (sdkTweenComponent.Removed)
+            {
+                sdkTweenComponent.Tweener.Kill();
+                ecsToCRDTWriter.DeleteMessage<PBTweenState>(sdkEntity);
+                tweenComponentPool.Release(sdkTweenComponent);
+                World.Remove<TweenComponent>(entity);
+                return;
+            }
```

Comment:
The `removed` property approach comes from the old explorer, I think we could handle this without having that `removed` property.

Probably you can have all  of these cleanup lines in a method and call it at `FinalizeComponents()` and at `HandleEntityDestruction()`... and probably at `HandleComponentRemoval()` also.

Since `HandleComponentRemoval()` is now at `TweenLoaderSystem` I think it would be OK to move that method from there into this other `TweenUpdaterSystem`.

Also you should be removing the custom components (AKA non protobuf/pb components) from the entity during "component removal", like these other systems do:
* [Example 1](https://github.com/decentraland/unity-explorer/blob/725bd825b0fd76b1cbe38bb0ae10d12ca14a12cc/Explorer/Assets/DCL/SDKComponents/AudioSources/Systems/CleanUpAudioSourceSystem.cs#L34)
* [Example 2](https://github.com/decentraland/unity-explorer/blob/725bd825b0fd76b1cbe38bb0ae10d12ca14a12cc/Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundReleaseSystem.cs#L34)

What do you think?

Reply:
nice! sounds good!